samples/opengl/depthMapTriangulation/image_triangulator.cpp

```diff
@@ -0,0 +1,949 @@
+                resolve_dependencies(mesh, unprojected_image, Point2D(y + step_small, x + step_med + step_med),
+                                     selected_vertices);  // right
+                resolve_dependencies(mesh, unprojected_image, Point2D(y + step_med, x + step_med + step_small),
+                                     selected_vertices);  // down
+            }
+        }
+    }
+}
+
+
+// ----------- actual triangulation -----------
+
+void Saiga::RQT_Triangulator::triangulate_RQT_selected_vertices(
+    MyMesh& current_mesh, ImageView<OpenTriangleMesh::VertexHandle> const& selected_vertices)
+{
+    // heap for quads that need to be triangulated
+    // consists of the coordinates of the left upper corner, the side_length of the quad and a bool indicating the
+    // triangulation (false: top left to bottom right, true: top right to bottom left)
+    std::vector<std::tuple<Point2D, int, bool>> quad_heap;
+    quad_heap.push_back(std::tuple<Point2D, int, bool>(Point2D(0, 0), RQT_side_len, false));
+
+    // flags to remember which parts of a quad are handled already. numeration starts with top left and goes in counter
+    // clock wise order
+    bool child_1, child_2, child_3, child_4;
+
+    MyMesh::VertexHandle curr_center_vh;
+
+    int curr_y, curr_x;
+    int curr_max_y, curr_max_x;
+    int curr_side_len, next_side_len, step_small, step_mid;
+    bool curr_triangulation;  // false: top left to bottom right, true: top right to bottom left
+    bool up_exists, left_exists, down_exists, right_exists;  // flags to help determine the correct triangulation
+    while (!quad_heap.empty())
+    {
+        // get current data
+        curr_y             = std::get<0>(quad_heap.back()).first;
+        curr_x             = std::get<0>(quad_heap.back()).second;
+        curr_side_len      = std::get<1>(quad_heap.back());
+        curr_max_y         = curr_y + curr_side_len - 1;
+        curr_max_x         = curr_x + curr_side_len - 1;
+        step_mid           = curr_side_len / 2;
+        step_small         = step_mid / 2;
+        next_side_len      = step_mid + 1;
+        curr_triangulation = std::get<2>(quad_heap.back());
+        quad_heap.pop_back();
+
+        // if the current quad is not part of the image it can be discarded
+        if (curr_y >= settings.image_height || curr_x >= settings.image_width)
+        {
+            continue;
+        }
+
+        // if the right-most lowest-most vertex is not within the image, the whole quad can be split without thought
+        if (curr_max_y >= settings.image_height || curr_max_x >= settings.image_width)
+        {
+            if (curr_side_len > 3)
+            {
+                // the quads will be split since at least two of them are out of the images bounds and the others
+                // have to fulfill the quadtree restriction
+                quad_heap.push_back(std::tuple<Point2D, int, bool>(Point2D(curr_y, curr_x), next_side_len, false));
```

Comment:
Same here. It's kind of confusing that you use an (y,x) point instead of (x,y). Changing it everywhere is a lot of work and might introduce new bugs.

So maybe can you just rename the `Point2D` type to `Point2DYX` or something?